src/vss_tools/model.py

```diff
@@ -277,6 +281,31 @@ def check_datatype_matching_allowed_unit_datatypes(self) -> Self:
             ), f"'{self.datatype}' is not allowed for unit '{self.unit}'"
         return self
 
+    @model_validator(mode="after")
+    def check_datatype_pattern(self) -> Self:
+        """
+        Checks that regular expression datatype 'pattern' field is:
+
+        1. string only
+        2. defined only for string typed nodes i.e., STRING and STRING_ARRAY
+        3. if default value is provided it is matching the specified pattern
+        """
+        if self.pattern:
+            assert bool(type(self.pattern) is str), f"Specified pattern: '{self.pattern}' must be a string"
+
+            # Datatypes.TUPLE[0] is the string name of the type.
+            allowed_for = f"It is allowed only for '{Datatypes.STRING[0]}' and '{Datatypes.STRING_ARRAY[0]}' types."
+            assert bool(
+                Datatypes.get_type(self.datatype) in [Datatypes.STRING, Datatypes.STRING_ARRAY]
+            ), f"Field 'pattern' is not allowed for '{self.fqn}' of type: '{self.datatype}'. {allowed_for}"
```

Comment:
The `fqn` info is already in the rendered exception and does not need to be given.

Reply:
I will update the error response.

Reply:
Done